fastlane/lib/fastlane/actions/device_grid/device_grid.rb

```diff
@@ -0,0 +1,104 @@
+# rubocop:disable Style/IndentationConsistency
+# We disable this rule since we intend opening HTML labels
+module Danger
+  class Dangerfile
+    module DSL
+      # A danger plugin: https://github.com/danger/danger
+      class DeviceGrid < Plugin
+        # @param languages: Array of languages you want to see (e.g. [en, de])
+        # @param devices: Array of deviecs you want to see (e.g. ["iphone4s", "ipadair"])
+        # @param prefix_command: Prefix the `fastlane run appetize_viewing_url_generator` command with something
+        #   this can be used to use `bundle exec`
+        def run(languages: nil, devices: nil, prefix_command: nil)
+          public_key = nil
+          [".fastlane/appetize_public_key.txt", "fastlane/appetize_public_key.txt"].each do |current| # not using Dir[...] since it doesn't show hidden files
+            public_key = File.read(current).strip if File.exist?(current)
+          end
+          raise("No public_key file found, make sure to run fastlane with `generate_device_grid` before calling `device_grid` in danger") if public_key.to_s.length == 0
+
+          devices ||= %w(iphone4s iphone5s iphone6s iphone6splus ipadair)
+          languages ||= ["en"]
+          prefix_command = "bundle exec" if File.exist?("Gemfile")
+          prefix_command ||= ""
+
+          deep_link_matches = pr_body.match(/:link:\s(.*)/) # :link: emoji
+          deep_link = deep_link_matches[1] if deep_link_matches
+
+          ENV["FASTLANE_DISABLE_COLORS"] = "true" # since we fetch the URL from the output
```

Comment:
is there a risk of this breaking some kind of environment setup ?

Isn't it better to 
1. save the current value
2. clean up when done.

That way you don't have to remove it in your test at the end of this PR